index.bs

```diff
@@ -1598,286 +1600,161 @@ A URL for Blob and File reference</h2>
-web developers are not likely to interact with all the headers,
-but the {{XMLHttpRequest/getAllResponseHeaders()}} method of {{XMLHttpRequest}}, if used,
-will show relevant [=/response=] headers.
+1. [=Assert=]: |url|'s [=url/scheme=] is <code>"blob"</code>.
+1. Let |entry| be the result of [=Blob URL/resolve|resolving=] |url|.
+1. If |entry| is not undefined, return |entry|'s [=Blob URL Entry/environment=]'s [=environment settings object/origin=].
+1. Let |nested url| be the result of [=URL parser|parsing=] |url|'s [=url/path=][0].
+1. Return a new [=opaque origin=], if |nested url| is failure, and |nested url|'s [=url/origin=] otherwise.
 
-<div class="example">
-  [=/Requests=] could look like this:
+</div>
 
-  <pre>
-  GET http://example.org:8080/550e8400-e29b-41d4-a716-446655440000
-  </pre>
+Note: The effect of this algorithm is that the origin of a Blob URL is always
+the same as that of the environment that created the URL,
+as long as the URL hasn't been revoked yet.
+If the URL was revoked the serialization of the origin will still remain the same
+as the serialization of the origin of the environment that created the Blob URL,
+but for opaque origins the origin itself might be distinct.
+This difference isn't observable though, since a revoked Blob URL can't be resolved/fetched anymore anyway.
 
-  If the {{Blob}} has an affiliated media type [[!RFC2046]] represented by its {{Blob/type}} attribute,
-  then the response message should include the Content-Type header from RFC7231 [[RFC7231]].
-  See [[#processing-media-types]].
+Issue: The [[URL]] spec should be updated to refer to this algorithm to resolve the origin of a Blob URL when the URL is first parsed.
+This is tracked in <a href="https://github.com/w3c/FileAPI/issues/63">issue #63</a> and in
+<a href="https://github.com/whatwg/url/issues/127">whatwg/url#127</a>.
 
-  Example response:
+<h4 id="lifeTime" dfn for="blob url" lt="lifetime|lifetime stipulation" export>
+Lifetime of Blob URLs</h3>
 
-  <pre>
-  200 OK
-  Content-Type: image/jpeg
-  Content-Length: 21897
+This specification extends the [=unloading document cleanup steps=] with the following steps:
 
-  ....
-  </pre>
+1. Let |environment| be the {{Document}}'s [=relevant settings object=].
+1. Let |store| be the user agent's [=Blob URL Store=];
+1. Remove from |store| any entries for which the [=map/value=]'s [=Blob URL Entry/environment=] is equal to |environment|.
 
-  If there is a <a>file read error</a> associated with the {{Blob}},
-  then a user agent acts as if a <a>network error</a> has occurred.
-</div>
+Issue: This needs a similar hook when a worker is unloaded.
 
 <h3 id="creating-revoking">
 Creating and Revoking a Blob URL</h3>
 
 <a>Blob URLs</a> are created and revoked using methods exposed on the {{URL}} object,
-supported by global objects {{Window}} [[HTML]] and {{WorkerGlobalScope}} [[Web Workers]].
+supported by global objects {{Window}} and {{WorkerGlobalScope}}.
```

Comment:
Is this "supported by...." sentence fragment even necessary? 

And maybe "static methods" as used above?